azafea/event_processors/endless/metrics/events/__init__.py

```diff
@@ -555,6 +555,110 @@ def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
         }
 
 
+class ParentalControlsBlockedFlatpakInstall(SingularEvent):
+    __tablename__ = 'parental_controls_blocked_flatpak_install'
+    __event_uuid__ = '9d03daad-f1ed-41a8-bc5a-6b532c075832'
+    __payload_type__ = 's'
+
+    app = Column(Unicode, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return {
+            'app': payload.get_string(),
+        }
+
+
+class ParentalControlsBlockedFlatpakRun(SingularEvent):
+    __tablename__ = 'parental_controls_blocked_flatpak_run'
+    __event_uuid__ = 'afca2515-e9ce-43aa-b355-7663c770b4b6'
+    __payload_type__ = 's'
+
+    app = Column(Unicode, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return {
+            'app': payload.get_string(),
+        }
+
+
+class ParentalControlsChanged(SingularEvent):
+    __tablename__ = 'parental_controls_changed'
+    __event_uuid__ = '449ec188-cb7b-45d3-a0ed-291d943b9aa6'
+    __payload_type__ = 'aa{sv}'
+
+    # One a{sv} per user, containing the fields from
+    # `com.endlessm.ParentalControls.AppFilter`.
+    app_filters = Column(JSONB, nullable=False)
+
+    @staticmethod
+    def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        return {'app_filters': [get_asv_dict(user_payload) for user_payload in get_child_values(payload)]}
```

Comment:
Yes, I think it would be better to change the malcontent code to submit one event per user, and only if their parental controls have actually changed; and pick out the keys into distinct fields in the same way as `ParentalControlsInitialSetup` does.

I wrote a whole spiel about how Metabase does not have native support for JSONB columns (take a look at the (poorly-named) "Location Event" table on https://metabase.dev.endlessm-sf.com/ and see that it's just treated as a string column by the GUI query editor), and then realised that even `ParentalControlsInitialSetup` is mostly jsonb and array (which metabase doesn't natively support either) so one will be working with SQL anyway. Still, I think easier to work with one less dimension.

Reply:
Done. As suggested and discussed on Slack, I folded the `ParentalControlsInitialSetup` event into `ParentalControlsChanged` using a new `IsInitialSetup` `a{sv}` field. This reduces the server-side code, makes the data format a bit more queryable, and simplifies the server-side code.